File: internal/model/blog.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

// BlogStatus 博客状态
type BlogStatus int32

const (
	BlogStatusDraft     BlogStatus = 0
	BlogStatusPublished BlogStatus = 1
	BlogStatusDeleted   BlogStatus = 2
)

// PrivacyLevel 隐私级别
type PrivacyLevel int32

const (
	PrivacyPublic   PrivacyLevel = 0
	PrivacyFriends  PrivacyLevel = 1
	PrivacyPrivate  PrivacyLevel = 2
)

// Blog 博客模型
type Blog struct {
	ID          string       `bson:"_id,omitempty" json:"id"`
	UserID      string       `bson:"user_id" json:"user_id"`
	Title       string       `bson:"title" json:"title"`
	Content     string       `bson:"content" json:"content"`
	Summary     string       `bson:"summary" json:"summary"`
	CoverImage  string       `bson:"cover_image" json:"cover_image"`
	Tags        string       `bson:"tags" json:"tags"`
	Category    string       `bson:"category" json:"category"`
	Privacy     PrivacyLevel `bson:"privacy" json:"privacy"`
	Status      BlogStatus   `bson:"status" json:"status"`
	CreatedAt   time.Time    `bson:"created_at" json:"created_at"`
	UpdatedAt   time.Time    `bson:"updated_at" json:"updated_at"`

	// 统计信息
	ViewsCount     int32 `bson:"views_count" json:"views_count"`
	LikesCount     int32 `bson:"likes_count" json:"likes_count"`
	CommentsCount  int32 `bson:"comments_count" json:"comments_count"`
	FavoritesCount int32 `bson:"favorites_count" json:"favorites_count"`
}

// BeforeInsert 插入前钩子
func (b *Blog) BeforeInsert(tx interface{}) {
	if b.ID == "" {
		b.ID = bson.NewObjectID().Hex()
	}
	now := time.Now()
	if b.CreatedAt.IsZero() {
		b.CreatedAt = now
	}
	b.UpdatedAt = now
}

// BeforeUpdate 更新前钩子
func (b *Blog) BeforeUpdate() {
	b.UpdatedAt = time.Now()
}

// CollectionName 集合名
func (Blog) CollectionName() string {
	return "blogs"
}

File: front/src/api/blog.ts
import apiClient from './client'
import { UserInfo } from './auth'

export interface Blog {
  id: string
  user_id: string
  title: string
  content: string
  summary: string
  cover_image: string
  tags: string[]
  category: string
  privacy: number
  status: number
  created_at: string
  updated_at: string
  views_count: number
  likes_count: number
  comments_count: number
  author?: UserInfo
}

export interface CreateBlogRequest {
  title: string
  content: string
  summary?: string
  cover_image?: string
  tags?: string[]
  category?: string
  privacy?: number
  status?: number
}

export interface BlogListResponse {
  blogs: Blog[]
  pagination: {
    page: number
    page_size: number
    total: number
    total_pages: number
  }
}

export const blogApi = {
  list: (params?: { page?: number; page_size?: number; category?: string; user_id?: string }) =>
    apiClient.get<BlogListResponse>('/blogs', { params }),
  
  get: (id: string) =>
    apiClient.get<{ blog: Blog }>(`/blogs/${id}`),
  
  create: (data: CreateBlogRequest) =>
    apiClient.post<{ blog: Blog }>('/blogs', data),
  
  update: (id: string, data: Partial<CreateBlogRequest>) =>
    apiClient.put<{ blog: Blog }>(`/blogs/${id}`, data),
  
  delete: (id: string) =>
    apiClient.delete(`/blogs/${id}`),
  
  listByUser: (userId: string, page?: number, page_size?: number) =>
    apiClient.get<BlogListResponse>(`/users/${userId}/blogs`, { params: { page, page_size } }),
}
